File: 228-SummaryRanges/SummaryRanges.py
class Solution(object):
    def summaryRanges(self, nums):
        """
        :type nums: List[int]
        :rtype: List[str]
        """
        if len(nums) == 0:
            return []
        results = []
        left_num = nums[0]
        right_num = nums[0]
        prev_num = nums[0]
        for num in nums[1:]:
            if prev_num + 1 == num:
                right_num = num
            else:
                if left_num >= right_num:
                    results.append(str(left_num))
                else:
                    results.append(str(left_num) + "->" + str(right_num))
                left_num = num
            prev_num = num
        if left_num >= right_num:
            results.append(str(left_num))
        else:
            results.append(str(left_num) + "->" + str(right_num))
        return results

File: 228-SummaryRanges/SummaryRanges.cpp
class Solution {
public:
    vector<string> summaryRanges(vector<int>& nums) {
        if (nums.size() == 0) {
            return {};
        }
        vector<string> results;
        int left_num = nums[0];
        int right_num = nums[0];
        int prev_num = nums[0];
        for (int x = 1; x < nums.size(); x++) {
            if (prev_num + 1 == nums[x]) {
                right_num = nums[x];
            }
            else {
                if (left_num >= right_num) {
                    results.push_back(to_string(left_num));
                }
                else {
                    results.push_back(to_string(left_num) + "->" + to_string(right_num));
                }
                left_num = nums[x];
            }
            prev_num = nums[x];
        }
        if (left_num >= right_num) {
            results.push_back(to_string(left_num));
        }
        else {
            results.push_back(to_string(left_num) + "->" + to_string(right_num));
        }
        return results;
    }
};
